### ISSUE TITLE
zstd support

### ISSUE THREAD

**Issue by ole-tange** [ckolivas/lrzip#61]
I am horribly impressed by (p)zstd https://github.com/facebook/zstd

The parallelized version is blazingly fast on a multicore system: You are getting gzip compression at lzo speed.

Will that be an option in addition to --zpaq --bzip2 --gzip --lzo --lzma?

**Comment by ckolivas** [comment]
No, as the rate limiting step with the faster compressors is the rzip first part. Speeding up the 2nd stage compression has no effect on overall speed.

**Comment by ckolivas** [comment]
Will reconsider for later version now as it will introduce an incompatibility issue with existing versions.

**Comment by pete4abw** [comment]
You can always do 
`lrzip -n`
and feed the output to zstd. IMHO adding another compressor that does not improve overall compression is regressive. In fact, using `lrzip -L4` is just about as fast as bzip and gzip and provides better compression  Remember lrzip's mission. See [These Test Comparison Results](https://github.com/pete4abw/lrzip/blob/lzma-sdk-19.00/doc/README.NEW.BENCHMARK.ALGO.md)

**Comment by mirh** [comment]
Idk about lrzip's mission, but it seems dumb to still make comparisons with bzip/deflate (and also probably PAQ, but at least that still hasn't been fully Pareto'd yet). 
The kernel is even [planning](https://www.phoronix.com/scan.php?page=news_item&px=Linux-Kernel-BZIP2-Dropping-RFC) to remove them now that zstd is a thing.

**Comment by ckolivas** [comment]
I said nothing about a mission.

**Comment by pete4abw** [comment]
> I said nothing about a mission.
I did. Poor choice of words. Intent perhaps w/b better. Provide better compression. Anyway, gzip and bzip2 are going nowhere. Adding a new method that is no better makes little sense. Zpaq made sense since it can be superior to lzma. JM2C

**Comment by mirh** [comment]
No better? I'm not sure I underlined enough how zstd is better in ***all*** metrics in ***all*** possible cases than bzip (maybe also deflate, but I cannot remember a graph covering all the spectrum of options there)

**Comment by ckolivas** [comment]
I guess you're missing his point as well. Neither Bzip2 nor deflate is used by default ever - at all - on lrzip, unless the user goes out of their way to choose it, and no one would today. Where I see it offering a use case is as an alternative to when users choose lzo for fast compression, zstd will offer similar speeds but better compression.

**Comment by pete4abw** [comment]
> No better? I'm not sure I underlined enough how zstd is better in _**all**_ metrics in _**all**_ possible cases than bzip (maybe also deflate, but I cannot remember a graph covering all the spectrum of options there)

I did a test run. It appears, at least for now that lrzip does not offer a big advantage for zstd over zstd alone.

Size | Filename | Time | Description
--- | --- | --- | ---
158270670 |  linux-5.10.tar.n.zst | 1:22 | zstd compressed after lrzip -n
160005503 | linux-5.10.tar.zst | 1:38 | zstd -17 -T0
897427042 | linux-5.10.tar.n | 0:12 | lrzip -n -S .n 
1158307840 |  linux-5.10.tar | N/A | Original File

I just don't see the value here.

**Comment by mirh** [comment]
Interesting. 
I wonder if that is also the case with `--fast`, `--ultra` or `--long`.

**Comment by pete4abw** [comment]
> Interesting.
> I wonder if that is also the case with `--fast`, `--ultra` or `--long`.

See above edits
